### DISCUSSION TITLE
stuck on Device Activation

### DISCUSSION THREAD

**Discussion by jonathanlermitage** [community/community#8468] [Copilot Conversations] [answered]
Copilot installed, I am redirected to GitHub to authenticate. Once authenticated (two-factor auth enabled with SMS), I am redirected to the "Device Activation" webpage, asking to "Enter the code displayed on your device" (8-digits), but I received nothing: no email (i checked my spams), no SMS.  
I tried this two times.

**Comment by jansorg** [comment, marked as answer]
The code is displayed in your IDE or editor, it's not a two-factor code.

**Reply by manuelduarte077** [reply]
Right there he leaves the code, he gets something like go and copy

**Reply by enisn** [reply]
Thank you for this information. I've been dealing with this for an hour

**Reply by enisn** [reply]
![image](https://user-images.githubusercontent.com/23705418/199432681-a9f489dd-0ae9-4d6d-84d9-9903ec709af1.png)
This UI doesn't explain anything. I think there should be a code in my authenticator app in my device.

**Reply by CyberMega8** [reply]
Thank you

**Reply by IshworKhadka** [reply]
Its copied to the clipboard.

**Reply by juancarlosortizr** [reply]
It's not copied to the clipboard. In my case, there's nothing in the clipboard.

**Comment by jonathanlermitage** [comment]
ho, ok, thx!

**Comment by Vipin-Sharma** [comment]
Thanks for the question @jonathanlermitage, I had the same problem today, an amazing tool with poor user experience :)

**Comment by DominikErnst** [comment]
OOOOOMMMMGGGG
You will find the Code in VS 2022 "Output" Window under "GitHub Copilot". You have to install the Extension first!
**_A device activation code XXXX-XXXX has been copied to the clipboard._**

**Reply by fernandezja** [reply]
"Device Activation" for Github Copilot to VS2022 is located in the Ouput window
(I copy a example in image)
![image](https://user-images.githubusercontent.com/201305/172051156-8975c72d-278d-4bfc-be0b-205c9079ea36.png)

**Reply by manuelduarte077** [reply]
nice

**Reply by ozt-j** [reply]
the same issues,  I can not receive Device Actication when I run "gh auth login" in MacOS Terminal, and I check my email many times, my editor is emacs.

**Reply by mikes-gh** [reply]
Late reply but the I found the code didnt appear in mac terminal (zsh).  I used powershell and there is a code that you can copy before opening the browser

**Reply by jianminPei** [reply]
thank you.I got it.

**Reply by igeoorge3g** [reply]
Thanks!

**Comment by wahyusa** [comment]
I got same condition, the marked answer has gave me new insight but since I am using Sublime Text LSP-copilot Package (Copilot in Sublime Text) I can just paste the code because I didn't get any notification about the code anywhere 😃 

The **code is already copied to clipboard automatically** just paste and done.

I am using Windows.

**Reply by vmklso** [reply]
wow , you are right!! , this is unexpected，while I am using the vs code

**Comment by helbing** [comment]
I use `gh auth login`, and the code is shown in the terminal:

<img width="643" alt="iShot_2023-04-18_21 51 50" src="https://user-images.githubusercontent.com/7275620/232798794-a8dc2b75-3f85-45a1-8411-d9b2edbc0025.png">
